test/lib/ansible_test/_util/target/pytest/plugins/ansible_pytest_unit_test_module_mocks.py

```diff
@@ -0,0 +1,384 @@
+                    run_module()
+                except _SyntheticSystemExit as sys_exit_exc:
+                    return sys_exit_exc.return_value
+                # pylint: disable-next=broad-except
+                except BaseException as base_exc:
+                    extra_advice = ''
+                    if isinstance(base_exc, ImportError):
+                        extra_advice = (
+                            '\n\n'
+                            'This may be caused by an incorrect '
+                            'importable path specification in the test.'
+                        )
+
+                    return pytest.fail(  # `return` is a hack for pylint
+                        'Invoking `{fqcn!s}` caused an unhandled '
+                        'error {exc!r}.{extra_advice!s}'.
+                        format(
+                            fqcn=name_or_fqcn, exc=base_exc,
+                            extra_advice=extra_advice,
+                        ),
+                    )
+                else:
+                    return pytest.fail(  # `return` is a hack for pylint
+                        'The module has not communicated '
+                        'a successful exit or a failure',
+                    )
+        return run_module_with
+    return make_module_wrapper
+
+
+@pytest.fixture
+def make_ansible_module_subprocess_caller():
+    """Create a subprocess-based module callable maker."""
+    def make_module_wrapper(name_or_fqcn):
+        def run_module_with(**module_kwargs):
+            module_args_defaults = {
+                '_ansible_keep_remote_files': False,
+                '_ansible_remote_tmp': '/tmp',
+            }
+
+            mod_args = module_kwargs.get('ANSIBLE_MODULE_ARGS', module_kwargs)
+            mod_args = dict(module_args_defaults, **mod_args)
+
+            args = json.dumps({'ANSIBLE_MODULE_ARGS': mod_args})
+            args = to_bytes(args, errors='surrogate_or_strict')
+
+            process = subprocess.Popen(  # pylint: disable=consider-using-with  # no CM @ py2
+                (sys.executable, '-m', name_or_fqcn),
+                stdin=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+            )
+
+            try:
+                stdoutdata, _stderrdata = process.communicate(input=args)
+            finally:
+                process.stdout.close()
+                process.stderr.close()
+                process.stdin.close()
+                wait_kwargs = {} if PY2 else {'timeout': 5}
+                process.wait(**wait_kwargs)
+
+            module_return_json_string = stdoutdata.decode()
+            return json.loads(module_return_json_string)
+        return run_module_with
+    return make_module_wrapper
+
+
+@pytest.fixture
+def ansible_module_args(mocker, monkeypatch, request):
+    """Patch and return stdin buffer with module args."""
+    monkeypatch.setattr(ansible.module_utils.basic, '_ANSIBLE_ARGS', None)
```

Comment:
@mattclay do you think it's fine to do such sort of monkey-patching private variables in an external collection? This would effectively make this constant a public API.

Reply:
See my previous [comment](https://github.com/ansible/ansible/pull/66430/files#r1043956789).